### PR TITLE
New Continuous Integration approach

### PR DESCRIPTION
**Pull request by arm61**

### CODE REVIEW

**Comment by arm61** [comment]
This PR contains the following:

## Work done

- Easy\*App now also installs using `pip` via `hatch` build system. 
- If installed as:
    ```
    git clone git@github.com:easyScience/EasyReflectometryApp.git
    pip install .
    ```
    The application can run from anywhere with
    ```
    EasyReflectometry
    ```
    In the terminal. 
- Application builds on CI for Windows and macOS (20 minutes down to 15 minutes)
- Application also runs on Windows and macOS
- For Ubuntu-linux `snap` is used to build the application
- Installing from `pip` works on ubuntu but required the installation of `libxcb-xinerama0` from `apt`
- The online installer is only made when the push is to master
- There is a lot of documentation of the workflow and processes
- I have removed a lot of things that EasyR wasn't using (translations/tutorial videos), these may get readded in future.

**Comment by arm61** [comment]
https://github.com/easyScience/EasyReflectometryLib/pull/58

Needs to merge first, and then the `pyproject.toml` should be updated to use the `develop` branch.

File: docs/build.md [hunk 1]
```diff
@@ -0,0 +1,77 @@
+# Development Env
+
+The EasyReflectometry project consists of two packages; an application and a library. 
+To develop effectively for EasyReflectometry, we work on the `develop` branch of both and then periodically merge to `master` when a release is made.
```
Comment (wardsimon) [inline docs/build.md:4]: The default branch is `main` not `master`

**Comment by AndrewSazonov** [comment]
Trying to do this `git clone git@github.com:easyScience/EasyReflectometryApp.git`

I got the following error:
```
Cloning into 'EasyReflectometryApp'...
git@github.com: Permission denied (publickey).
fatal: Could not read from remote repository.

Please make sure you have the correct access rights
and the repository exists.

```

**Review by wardsimon — COMMENTED** [review]
I could only find one minor ?mistake? Looks good 👍

File: docs/build.md (hunk 1 above)
Comment (arm61) [reply docs/build.md:4]:
```suggestion
To develop effectively for EasyReflectometry, we work on the `develop` branch of both and then periodically merge to `main` when a release is made. 
```

File: docs/build.md [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+
+The EasyReflectometry project consists of two packages; an application and a library. 
+To develop effectively for EasyReflectometry, we work on the `develop` branch of both and then periodically merge to `main` when a release is made. 
+
+With this in mind, the best way to build a development environment for EasyReflectometry is as follows. 
+
+```console
+$ mamba create -y -n easy_refl python=3.9
+$ mamba activate easy_refl
+$ git clone git@github.com:easyScience/EasyReflectometryLib.git
```
Comment (arm61) [inline docs/build.md:11]:
```suggestion
$ git clone https://github.com/easyScience/EasyReflectometryLib.git
```

File: docs/build.md [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+to use the directory where you have cloned `EasyReflectometryLib`, i.e.
+
+```toml
+'EasyReflectometryLib @ file:///path/to/EasyReflectometryLib',
+```
+
+With this modification in place, you should then be able to install `EasyReflectometryApp`
+
+```console
+$ git clone git@github.com:easyScience/EasyReflectometryApp.git
```
Comment (arm61) [inline docs/build.md:33]:
```suggestion
$ git clone https://github.com/easyScience/EasyReflectometryApp.git
```

File: docs/build.md (hunk 2 above)
Comment (arm61) [reply docs/build.md:11]: @AndrewSazonov

File: docs/build.md (hunk 3 above)
Comment (AndrewSazonov) [reply docs/build.md:33]: You should also be able to install it via `pip` directly from `GitHub` even without cloning:
`pip install git+https://github.com/easyScience/EasyReflectometryApp`. 

And one can also select any specific branch like this: 
`pip install git+https://github.com/easyScience/EasyReflectometryApp@new_ci`

Comment (arm61) [reply docs/build.md:33]:
```suggestion
$ git clone git+https://github.com/easyScience/EasyReflectometryApp.git
```

File: docs/build.md (hunk 2 above)
Comment (arm61) [reply docs/build.md:11]:
```suggestion
$ git clone https://github.com/easyScience/EasyReflectometryLib.git
```

**Comment by rozyczko** [comment]
Current build snap, installed on Ubuntu 20.04 doesn't show charts at all, due to already known issue with it being sandboxed.

> ERROR:platform_shared_memory_region_posix.cc(249)] Creating shared memory in /dev/shm/.org.chromium.Chromium.F2Jffx failed: Permission denied (13)

I guess once we have properly signed/notarized snaps, this will be moot?

**Comment by arm61** [comment]
> Current build snap, installed on Ubuntu 20.04 doesn't show charts at all, due to already known issue with it being sandboxed.
> 
> > ERROR:platform_shared_memory_region_posix.cc(249)] Creating shared memory in /dev/shm/.org.chromium.Chromium.F2Jffx failed: Permission denied (13)
> 
> I guess once we have properly signed/notarized snaps, this will be moot?

@wardsimon ?

**Comment by wardsimon** [comment]
This should now be up and running, ready to merge

**Pull request merged**